frontend/src/components/nav/navbar.js

```diff
@@ -112,9 +122,12 @@ class NavBar extends React.Component {
                             onKeyUp={this.getSearchInput}   
                             value={this.state.word}
                             onChange={this.update()}
-                            onKeyPress={this.searchFunc()}
+                            // onKeyPress={this.searchFunc()}
+                            onKeyPress={this.debounceSearch(this.searchFunc(), 250)}
                          />
-                        <FontAwesomeIcon icon={faSearch} className={this.state.iconClassName} onClick={() => this.startSearch(this.state.word)} />
+                        
+                        {/* <FontAwesomeIcon icon={faSearch} className={this.state.iconClassName} onClick={() => this.startSearch(this.state.word)} /> */}
+                        <FontAwesomeIcon icon={faSearch} className={this.state.iconClassName} onClick={() => this.debounceSearch(this.startSearch(this.state.word), 250)} />
```

Comment:
this.startSearch(this.state.word) will invoke this.startSearch immediately. To pass this.startSearch as an argument, do this instead:

this.debounceSearch(this.startSearch, 250)

Also, this is not setting this.startSearch with debounce as the event handler. Try this instead:

onClick={this.debounceSearch(this.startSearch, 250)}